modules/300-prometheus/hooks/metrics_web_interfaces.go

```diff
@@ -147,13 +150,20 @@ func filterIngress(obj *unstructured.Unstructured) (go_hook.FilterResult, error)
 func domainMetricHandler(input *go_hook.HookInput) error {
 	snap := input.Snapshots["ingresses"]
 	input.MetricsCollector.Expire("deckhouse_exported_domains")
+	globalHTTPSMode := input.ConfigValues.Get("global.modules.https.mode").String()
 
 	for _, sn := range snap {
 		if sn == nil {
 			continue
 		}
 
 		domain := sn.(exportedWebInterface)
+
+		if globalHTTPSMode == "OnlyInURI" {
+			re := regexp.MustCompile(`^http://`)
+			domain.URL = re.ReplaceAllString(domain.URL, "https://")
```

Comment:
Do we need regular expressions here or can we handle it by using `strings.Replace` / `strings.ReplaceAll`?

Reply:
Fixed.